agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -105,6 +106,38 @@ namespace beerocks {
 #define FSM_IS_IN_STATE(eState) (m_eFSMState == EState::eState)
 #define FSM_CURR_STATE_STR s_arrStates[int(m_eFSMState)]
 
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Helper Functions /////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+template <class SlavesContainer, class MAC>
+std::shared_ptr<backhaul_manager::sRadioInfo> findRadioInfo(const SlavesContainer &slave_container,
+                                                            const MAC &mac)
+{
+    // for each radio info in the container
+    //      for each associated client
+    //          find if there is an entry with the given mac within this associated client
+
+    for (const auto &slave : slave_container) {
+        if (!slave) {
+            continue;
+        }
+
+        LOG(DEBUG) << "looking into info for hostap iface: " << slave->hostap_iface;
```

Comment:
In this specific case, yes, it seems that the log may be removed.
As for the general case: [I happened](https://github.com/CppCon/CppCon2019/tree/master/Posters/yet_another_fast_log) to deal with logs in the past. So I _have_ a strong opinion on the matter ;-).
If I need to summarize it in few sentences it would be:
Like any other aspect of the supporting environment for the code, logs must be useful. For a log to be useful it needs to tell a story, it needs to guide you through the events happened in the system. You should have a background in the way the system is built but you shouldn't have a deep knowledge in the code itself. For example, [tech support](https://www.bmc.com/blogs/support-levels-level-1-level-2-level-3/) should be able to monitor the system using the logs after a short while working with it, yet without being able to read the code itself.
It is not easy, and it is a continues mission to keep the log file useful, clean, without repeated messages, searchable - useful. Too often, however, projects neglect the log though it sometimes the only tool for monitoring and debugging.
I have never seen a real discussion (except for those I initiated), that sets customers, targets, guidelines, etc. regarding the logs. Although, in each project, everyone has an opinion and everyone uses the logs.